tests/integration/test_consume_flow.py

```diff
@@ -0,0 +1,147 @@
+    file = FilesTypeFactory(file_dict)
+    files = [file]
+
+    # Encrypt file objects
+    encrypt_response = data_provider.encrypt(
+        files, "http://172.15.0.4:8030/api/services/encrypt"
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
+
+    # Set ERC20 Data
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            publisher_wallet.address,
+            publisher_wallet.address,
+            ZERO_ADDRESS,
+            get_address_of_type(config, "Ocean"),
+        ],
+        uints=[web3.toWei("100", "ether"), 0],
+        bytess=[b""],
+    )
+
+    # Publish a plain asset with one data token on chain
+    ddo = asset.create(
+        metadata=metadata,
+        publisher_wallet=publisher_wallet,
+        encrypted_files=encrypted_files,
+        erc721_address=erc721_address,
+        erc20_tokens_data=[erc20_data],
+    )
+
+    assert ddo, "The asset is not created."
+    assert ddo.nft["name"] == "NFT1"
+    assert ddo.nft["symbol"] == "NFTSYMBOL"
+    assert ddo.nft["address"] == erc721_address
+    assert ddo.nft["owner"] == publisher_wallet.address
+    assert ddo.datatokens[0]["name"] == "Datatoken 1"
+    assert ddo.datatokens[0]["symbol"] == "DT1"
+
+    service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    erc20_token = ERC20Token(web3, ddo.datatokens[0]["address"])
+
+    # Mint 50 ERC20 tokens in consumer wallet from publisher. Max cap = 100
+    erc20_token.mint(
+        account_address=consumer_wallet.address,
+        value=to_wei(50),
+        from_wallet=publisher_wallet,
+    )
+
+    # Initialize service
+    response = data_provider.initialize(
+        did=ddo.did,
+        service_id=service.id,
+        file_index=0,
+        consumer_address=consumer_wallet.address,
+        service_endpoint="http://172.15.0.4:8030/api/services/initialize",
+    )
+    assert response
+    assert response.status_code == 200
+
+    # Start order for consumer
+    tx_id = erc20_token.start_order(
+        consumer=consumer_wallet.address,
+        service_id=int(service.id),
+        provider_fees=get_provider_fees(),
+        from_wallet=consumer_wallet,
+    )
+    # Download file
+    destination = config.downloads_path
+    if not os.path.isabs(destination):
+        destination = os.path.abspath(destination)
+    if not os.path.exists(destination):
+        os.mkdir(destination)
+
+    for i in range(len(files)):
+        response = data_provider.download(
+            did=ddo.did,
+            service_id=service.id,
+            tx_id=tx_id,
+            file_index=i,
+            consumer_wallet=consumer_wallet,
+            download_endpoint="http://172.15.0.4:8030/api/services/download",
+            destination_folder=destination,
+        )
+        assert response
+        assert response.status_code == 200
+
+    assert len(os.listdir(destination)) >= 1, "The asset folder is empty."
```

Comment:
please make this assertion tighter: count the files before and after downloading, and check that the difference is == to the number of files.

Reply:
== len(files)